### PR TITLE
Reverse a Linked List recursively

### PR DESCRIPTION
**Pull request by thesagarpasrija**
c++ implementation to reverse a linked list recursively by reversing the pointers to nodes of linked list

### CODE REVIEW

File: code/data_structures/linked_list/linked_lists_operations/Reverse_recursion_linked_list.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
+#include<iostream>
+#include<stack>
```
Comment (arnavb) [inline code/data_structures/linked_list/linked_lists_operations/Reverse_recursion_linked_list.cpp]: No need to include this if you don't plan on using it. 😝

File: code/data_structures/linked_list/linked_lists_operations/Reverse_recursion_linked_list.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+void insertInMiddle(node*&head,int d,int p){
+    if(p==0){
+        insertAtHead(head,d);
+    }
+    else{
+        //assuming p<=length of LL
+        int jump=1;
+        node*temp=head;
+        while(jump<=p-1){
```
Comment (arnavb) [inline code/data_structures/linked_list/linked_lists_operations/Reverse_recursion_linked_list.cpp]: Use the condition `< p` instead. While yours is equivalent, it is also unconventional. Also, this loop could be converted into a for loop.

File: code/data_structures/linked_list/linked_lists_operations/Reverse_recursion_linked_list.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+}
+istream& operator>>(istream&is, node*&head){
+    takeInput(head);
+    return is;
+}
+
+
+///optimized recursive reverse
+node*recReverse2(node*head){
+    if(head==NULL || head->next==NULL){
```
Comment (arnavb) [inline code/data_structures/linked_list/linked_lists_operations/Reverse_recursion_linked_list.cpp]: C++, use `nullptr`, not `NULL`

**Comment by thesagarpasrija** [comment]
@arnavb @AdiChat did all the changes. Please review the code.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**